utils/utils.go

```diff
@@ -432,3 +437,57 @@ func techArrayToString(techsArray []coreutils.Technology, separator string) (res
 	}
 	return strings.Join(techString, separator)
 }
+
+type UrlAccessChecker struct {
+	connected *bool
+	waitGroup *errgroup.Group
```

Comment:
Will work without the pointer
```suggestion
	waitGroup errgroup.Group
```